[Cosmos] Hybrid Search query pipeline (#38275)
Azure/azure-sdk-for-python: sdk/cosmos/azure-cosmos/README.md

All of these mentioned queries would look something like this:

- `SELECT TOP 10 c.id, c.text FROM c WHERE FullTextContains(c.text, 'quantum')`

- `SELECT TOP 10 c.id, c.text FROM c WHERE FullTextContainsAll(c.text, 'quantum', 'theory')`

- `SELECT TOP 10 c.id, c.text FROM c WHERE FullTextContainsAny(c.text, 'quantum', 'theory')`

- `SELECT TOP 10 c.id, c.text FROM c ORDER BY RANK FullTextScore(c.text, ['quantum', 'theory'])`

- `SELECT TOP 10 c.id, c.text FROM c ORDER BY RANK RRF(FullTextScore(c.text, ['quantum', 'theory']), FullTextScore(c.text, ['model']))`

- `SELECT TOP 10 c.id, c.text FROM c ORDER BY RANK RRF(FullTextScore(c.text, ['quantum', 'theory']), FullTextScore(c.text, ['model']), VectorDistance(c.embedding, {item_embedding}))"`

These queries must always use a TOP or LIMIT clause within the query since hybrid search queries have to look through a lot of data otherwise and may become too expensive or long-running.
Since these queries are relatively expensive, the SDK sets a default limit of 1000 max items per query - if you'd like to raise that further, you

> can use the `AZURE_COSMOS_MAX_ITEM_BUFFER_HYBRID_SEARCH` environment variable to do so. However, be advised that queries with too many vector results

Reviewer: Any added env variables must have a programmatic equivalent to achieve the same thing.
This name also seems a bit arbitrary.... what's the "buffer" here? `AZURE_COSMOS_HYBRID_SEARCH_MAX_ITEMS` seems more to-the-point?

Author: yep agreed - I did this name to align with the one we used for vector search, but I agree this name makes more sense

may have additional latencies associated with searching in the service.

> You can find our sync samples [here][cosmos_index_sample] and our async samples [here][cosmos_index_sample_async] as well to help yourself out.

Reviewer: Maybe:
```suggestion
You can find our sync samples [here][cosmos_index_sample] and our async samples [here][cosmos_index_sample_async] as well for additional guidance.
```
